test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,266 @@
+    assert split_docs[0].meta["importance"] == "high"
+    assert split_docs[0].meta["custom_field"] == 123
+
+    # New metadata should be added
+    assert "header" in split_docs[0].meta
+    assert "split_id" in split_docs[0].meta
+
+
+# Error and edge case handling
+def test_non_text_document(caplog):
+    """Test that the component correctly handles non-text documents."""
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=None)]
+
+    # Should raise ValueError about text documents
+    with pytest.raises(ValueError, match="only works with text documents"):
+        splitter.run(documents=docs)
+
+
+def test_empty_document_list():
+    """Test handling of an empty document list."""
+    splitter = MarkdownHeaderSplitter()
+    result = splitter.run(documents=[])
+    assert result["documents"] == []
+
+
+def test_invalid_secondary_split_at_init():
+    """Test that an invalid secondary split type raises an error at initialization time."""
+    with pytest.raises(ValueError, match="split_by must be one of"):
+        MarkdownHeaderSplitter(secondary_split="invalid_split_type")
+
+
+def test_invalid_split_parameters_at_init():
+    """Test invalid split parameter validation at initialization time."""
+    # Test split_length validation
+    with pytest.raises(ValueError, match="split_length must be greater than 0"):
+        MarkdownHeaderSplitter(secondary_split="word", split_length=0)
+
+    # Test split_overlap validation
+    with pytest.raises(ValueError, match="split_overlap must be greater than or equal to 0"):
+        MarkdownHeaderSplitter(secondary_split="word", split_overlap=-1)
+
+
+def test_empty_content_handling():
+    """Test handling of documents with empty content."""
+    splitter_skip = MarkdownHeaderSplitter()  # skip empty documents by default
+    docs = [Document(content="")]
+    result = splitter_skip.run(documents=docs)
+    assert len(result["documents"]) == 0
+
+    splitter_no_skip = MarkdownHeaderSplitter(skip_empty_documents=False)
+    docs = [Document(content="")]
+    result = splitter_no_skip.run(documents=docs)
+    assert len(result["documents"]) == 1
+
+
+# Output format and split ID checks
+def test_document_splitting_format():
+    """Test that the format of split documents is correct."""
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content="# Header\nContent")]
+    result = splitter.run(documents=docs)
+
+    # Basic validation of the output structure
+    assert isinstance(result, dict)
+    assert "documents" in result
+    assert isinstance(result["documents"], list)
+
+
+def test_split_id_sequentiality_primary_and_secondary():
+    text = "# Header\n" + "Word " * 30
+    # Test primary splitting
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=text)]
+    result = splitter.run(documents=docs)
+    split_ids = [doc.meta["split_id"] for doc in result["documents"]]
+    assert split_ids == list(range(len(split_ids)))
```

Comment:
Could we make this test a bit more complicated? Running this locally this example text only produces one document so the number of split_ids is just one. It would be good to see if this works as expected when multiple documents are created like using your `sample_text`